### PR TITLE
Only persist session when icon update was user initiated

### LINKED ISSUE #158595: `xterm` is loaded even when not visible
* Open a folder
* Open terminal
* Hide panel
* Reload Window

=> 🐛 all `xterm` related JS files seem to be loaded into the workbench when checking network tab.

image

### PR DESCRIPTION
**Pull request by Tyriar**
Fixes #158595

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/browser/terminalInstance.ts [hunk 1]
```diff
@@ -1464,7 +1464,7 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 		}
 		if (originalIcon !== this.shellLaunchConfig.icon || this.shellLaunchConfig.color) {
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:1465]: won't the first condition here always be false?

`originalIcon !== this.shellLaunchConfig.icon` 

if i'm missing something and that could be true, could use `this._icon` instead to also capture `this._shellLaunchConfig.attachPersistentProcess`

**Review by meganrogge — CHANGES_REQUESTED** [review]
confused about when the icon will be changed by the process

EDIT:

I see that we fire this on startup, so it's not the process setting it, it's us

File: src/vs/workbench/contrib/terminal/browser/terminalInstance.ts (hunk 1 above)
Comment (Tyriar) [reply src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:1465]: It's not always false, I'm not sure the exact case where this happens but there's some complexity around deferred icon resolving/updating in the terminal tab.

**Comment by Tyriar** [comment]
> I see that we fire this on startup, so it's not the process setting it, it's us

It's us (vscode), not the user. We only want the terminal to persist if the terminal has had user interaction

**Review by meganrogge — APPROVED** [review]

**Pull request merged**